x/auth/client/cli/tx_sign.go

```diff
@@ -230,58 +231,49 @@ func makeSignCmd() func(cmd *cobra.Command, args []string) error {
 			return err
 		}
 
-		// if --signature-only is on, then override --append
-		generateSignatureOnly, _ := cmd.Flags().GetBool(flagSigOnly)
-		multisigAddrStr, _ := cmd.Flags().GetString(flagMultisig)
-
-		from, _ := cmd.Flags().GetString(flags.FlagFrom)
+		generateSignatureOnly, _ := f.GetBool(flagSigOnly)
+		multisigAddrStr, _ := f.GetString(flagMultisig)
+		from, _ := f.GetString(flags.FlagFrom)
 		_, fromName, err := client.GetFromFields(txFactory.Keybase(), from, clientCtx.GenerateOnly)
 		if err != nil {
 			return fmt.Errorf("error getting account from keybase: %w", err)
 		}
 
 		if multisigAddrStr != "" {
 			var multisigAddr sdk.AccAddress
-
 			multisigAddr, err = sdk.AccAddressFromBech32(multisigAddrStr)
 			if err != nil {
 				return err
 			}
-
 			err = authclient.SignTxWithSignerAddress(
 				txF, clientCtx, multisigAddr, fromName, txBuilder, clientCtx.Offline,
 			)
 			generateSignatureOnly = true
 		} else {
-			flagAppend, _ := cmd.Flags().GetBool(flagAppend)
-			appendSig := flagAppend && !generateSignatureOnly
-			if appendSig {
+			flagAppend, _ := f.GetBool(flagAppend)
+			if flagAppend || generateSignatureOnly {
```

Comment:
It's unclear to me why this is the logic we want hear. Was the prior behavior incorrect?

Reply:
Looking at this more closely... it doesn't seem that we're actually handling the `--append` flag properly at all.

The documentation of `authclient.SignTx` states:
```
// SignTx appends a signature to a transaction. If appendSig
// is false, it replaces the signatures already attached with the new signature.
```

Looking downstream of this, it appears as if we are always actually overwriting the signature (incorrect behavior as the default value for `--append` is true), and there's no reading of the `--append` flag anywhere else.

Can someone with more context confirm here? cc @amaurymartiny @aaronc